cmd/fleetctl/gitops.go

```diff
@@ -174,54 +213,122 @@ func gitopsCommand() *cli.Command {
 	}
 }
 
-func checkAppleBMDefaultTeam(config *spec.GitOps, fleetClient *service.Client) (
-	appleBMDefaultTeam string, appleBMDefaultTeamFound bool, err error,
+// checkABMTeamAssignments validates the spec, and finds if:
+//
+// 1. The user is using the legacy apple_bm_default_team config.
+// 2. All teams assigned to ABM tokens already exist.
+// 3. Performs validations according to the spec for both the new and the
+// deprecated key used for this setting.
+func checkABMTeamAssignments(config *spec.GitOps, fleetClient *service.Client) (
+	abmTeams []string, missingTeam bool, usesLegacyConfig bool, err error,
 ) {
 	if mdm, ok := config.OrgSettings["mdm"]; ok {
-		if mdmMap, ok := mdm.(map[string]interface{}); ok {
-			if appleBMDT, ok := mdmMap["apple_bm_default_team"]; ok {
-				if appleBMDefaultTeam, ok = appleBMDT.(string); ok {
-					teams, err := fleetClient.ListTeams("")
-					if err != nil {
-						return "", false, err
-					}
-					// Normalize AppleBMDefaultTeam for Unicode support
+		if mdmMap, ok := mdm.(map[string]any); ok {
+			appleBMDT, hasLegacyConfig := mdmMap["apple_bm_default_team"]
+			appleBM, hasNewConfig := mdmMap["apple_business_manager"]
+
+			if hasLegacyConfig && hasNewConfig {
+				return nil, false, false, errors.New(fleet.AppleABMDefaultTeamDeprecatedMessage)
+			}
+
+			if !hasLegacyConfig && !hasNewConfig {
+				return nil, false, false, nil
+			}
+
+			teams, err := fleetClient.ListTeams("")
+			if err != nil {
+				return nil, false, false, err
+			}
+			teamNames := map[string]struct{}{}
+			for _, tm := range teams {
+				teamNames[tm.Name] = struct{}{}
+			}
+
+			if hasLegacyConfig {
+				if appleBMDefaultTeam, ok := appleBMDT.(string); ok {
+					// normalize for Unicode support
 					appleBMDefaultTeam = norm.NFC.String(appleBMDefaultTeam)
-					for _, team := range teams {
-						if team.Name == appleBMDefaultTeam {
-							appleBMDefaultTeamFound = true
-							break
-						}
+					abmTeams = append(abmTeams, appleBMDefaultTeam)
+					usesLegacyConfig = true
+					if _, ok = teamNames[appleBMDefaultTeam]; !ok {
+						missingTeam = true
 					}
-					if !appleBMDefaultTeamFound {
-						// If team is not found, we need to remove the AppleBMDefaultTeam from the global config, and then apply it after teams are processed
-						mdmMap["apple_bm_default_team"] = ""
+				}
+			}
+
+			if hasNewConfig {
+				if settingMap, ok := appleBM.([]any); ok {
+					for _, item := range settingMap {
+						if cfg, ok := item.(map[string]any); ok {
+							for _, teamConfigKey := range []string{"macos_team", "ios_team", "ipados_team"} {
+								if team, ok := cfg[teamConfigKey].(string); ok && team != "" {
+									// normalize for Unicode support
+									team = norm.NFC.String(team)
```

Comment:
Is that something we have to do every time we want to compare team names? I wonder if there are other places where we have broken comparisons because we don't normalize unicode strings...